importer/DeHeDe.py

```diff
@@ -0,0 +1,90 @@
+from Monument import Monument, Dataset
+import importer_utils as utils
+import importer as importer
+
+
+class DeHeDe(Monument):
+
+    def set_location(self):
+        if self.has_non_empty_attribute("ortsteil"):
+            location_q = utils.q_from_wikipedia("de", self.ortsteil)
+            if location_q:
+                self.add_statement("location", location_q)
+
+    def set_adm_location(self):
+        city_q = utils.q_from_wikipedia("de", self.stadt)
+        self.add_statement("located_adm", city_q)
+
+    def update_descriptions(self):
+        placename = None
+        germany = {
+            "de": "Deutschland",
+            "en": "Germany",
+            "sv": "Tyskland"
+        }
+        templates = {
+            "en": "Kulturdenkmal in {}",
+            "de": "Kulturdenkmal in {}",
+            "sv": "kulturminnesmärke i {}"
+        }
+        for lg in templates:
+            country = germany[lg]
+            if self.has_non_empty_attribute("stadt"):
+                placename = utils.get_rid_of_brackets(self.stadt)
+                placename = "{}, {}".format(placename, country)
+            else:
+                placename = country
+            desc = templates[lg].format(placename)
+            self.add_description(lg, desc)
+
+    def update_labels(self):
+        if self.has_non_empty_attribute("bezeichnung"):
+            german = utils.remove_markup(self.bezeichnung)
+        elif self.has_non_empty_attribute("adresse"):
+            german = utils.remove_markup(self.adresse)
+        self.add_label("de", german)
+
+    def set_heritage_id(self):
+        self.add_statement("denkXweb", str(self.nummer))
+
+    def set_street_address(self):
+        street_with_no = utils.remove_markup(self.adresse)
+        town = utils.get_rid_of_brackets(self.stadt)
+        address = "{}, {}".format(street_with_no, town)
+        self.add_statement("located_street", address)
+
+    def set_monuments_all_id(self):
+        self.monuments_all_id = str(self.nummer)
+
+    def exists_with_monument_article(self, language):
+        return super().exists_with_monument_article("de", "artikel")
+
+    def __init__(self, db_row_dict, mapping, data_files, existing, repository):
+        Monument.__init__(self, db_row_dict, mapping,
+                          data_files, existing, repository)
+        self.set_monuments_all_id()
+        self.set_changed()
+        self.wlm_source = self.create_wlm_source(self.monuments_all_id)
+        self.set_source()
+        self.set_registrant_url()
```

Comment:
You add this but never seem to make use of it (per preview)?

Reply:
Per the comment [here](https://github.com/Vesihiisi/COH-tools/pull/82#pullrequestreview-60518441): Would a good general strategy be to:
* Do nothing with the regisrar_url if
** The url is already auto generated by the id property on Wikidata
** The url is invalid (no longer valid or was never valid)
* Add it as a P973 qualifier on either the id claim or the heritage status claim otherwise.
?

Reply:
Added this as a qualifier to heritage id.

Reply:
For Hessen it seems as though the rl is already auto generated by the id property on Wikidata. Which would fall in the "do nothing" category.

Do you think the decision method above works in general? IS there somewhere else where we would want to use the url?